pkg/strategy/support/strategy.go

```diff
@@ -439,7 +390,48 @@ func (s *Strategy) Run(ctx context.Context, orderExecutor bbgo.OrderExecutor, se
 	s.orderExecutor = orderExecutor
 
 	// StrategyController
-	s.status = types.StrategyStatusRunning
+	s.Status = types.StrategyStatusRunning
+
+	s.OnSuspend(func() error {
+		var err error = nil
+		// Cancel all order
+		for _, order := range s.orderStore.Orders() {
+			err = s.cancelOrder(order.OrderID, ctx, s.orderExecutor)
+		}
+		if err != nil {
+			errMsg := "Not all orders are cancelled! Please check again."
+			log.WithError(err).Errorf(errMsg)
+			s.Notify(errMsg)
+		} else {
+			s.Notify("All orders cancelled.")
+		}
+
+		// Save state
+		if err2 := s.SaveState(); err2 != nil {
+			log.WithError(err2).Errorf("can not save state: %+v", s.state)
+		} else {
+			log.Infof("%s position is saved.", s.Symbol)
+		}
+
+		return err
+	})
+
+	s.OnEmergencyStop(func() error {
+		// Close 100% position
+		percentage, _ := fixedpoint.NewFromString("100%")
```

Comment:
you can do: fixedpoint.NewFromtFloat(1.0)

Reply:
done